src/overlays/actors/ovl_En_Karebaba/z_en_karebaba.c

```diff
@@ -19,57 +37,480 @@ const ActorInit En_Karebaba_InitVars = {
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Karebaba/func_80A956D0.s")
+    if (this->actionFunc != EnKarebaba_Dead) {
+        if (this->actionFunc == EnKarebaba_Dying) {
+            Actor_MoveForward(&this->actor);
+            func_8002E4B4(globalCtx, &this->actor, 10.0f, 15.0f, 10.0f, 5);
+        } else {
+            func_8002E4B4(globalCtx, &this->actor, 0.0f, 0.0f, 0.0f, 4);
+            if (this->boundFloor == NULL) {
+                this->boundFloor = this->actor.floorPoly;
+            }
+        }
+        if (this->actionFunc != EnKarebaba_Dying && this->actionFunc != EnKarebaba_DeadItemDrop) {
+            if (this->actionFunc != EnKarebaba_Regrow && this->actionFunc != EnKarebaba_Grow) {
+                CollisionCheck_SetAT(globalCtx, &globalCtx->colChkCtx, &this->headCollider);
+                CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->bodyCollider);
+            }
+            CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->headCollider);
+            Actor_SetHeight(&this->actor, (this->actor.scale.x * 10.0f) / 0.01f);
+            height = this->actor.initPosRot.pos.y + 40.0f;
+            this->actor.posRot2.pos.x = this->actor.initPosRot.pos.x;
+            this->actor.posRot2.pos.y = CLAMP_MAX(this->actor.posRot2.pos.y, height);
+            this->actor.posRot2.pos.z = this->actor.initPosRot.pos.z;
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Karebaba/func_80A95748.s")
+void EnKarebaba_DrawCenterShadow(EnKarebaba* this, GlobalContext* globalCtx) {
+    MtxF matrix;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Karebaba/func_80A957E0.s")
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_en_karebaba.c", 1013);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Karebaba/func_80A95860.s")
+    func_80094044(globalCtx->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Karebaba/func_80A958EC.s")
+    gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, 0, 0, 0, 255);
+    func_80038A28(this->boundFloor, this->actor.initPosRot.pos.x, this->actor.initPosRot.pos.y, this->actor.initPosRot.pos.z, &matrix);
+    Matrix_Mult(&matrix, MTXMODE_NEW);
+    Matrix_Scale(0.15f, 1.0f, 0.15f, MTXMODE_APPLY);
+    gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_en_karebaba.c", 1029), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gSPDisplayList(POLY_XLU_DISP++, D_04049210);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Karebaba/func_80A95940.s")
+    CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_en_karebaba.c", 1034);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Karebaba/func_80A959D8.s")
+void EnKarebaba_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    static Color_RGBA8 sColorBlack = { 0, 0, 0, 0 };
+    static Gfx* sDisplayLists[] = { D_06001330, D_06001628, D_06001828 };
+    static Vec3f sVecZero2 = { 0.0f, 0.0f, 0.0f };
```

Comment:
same comment as above, you can remove the `s` prefix from these variables, and id call it zeroVec again